scripts/generate-go-crd-clients/generate-types-file.go

```diff
@@ -344,18 +348,22 @@ func isResourceReference(d fielddesc.FieldDescription) bool {
 	return false
 }
 
+func isSecretReference(d fielddesc.FieldDescription) bool {
+	return d.ShortName == "secretKeyRef"
```

Comment:
Could you add a bit more check here on `d.Children`? ex: If there are two children with name "name" and "key", then we know for sure it is a secretReference. This avoids us mistakenly treat cases like below as secretReference.

https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/crds/run_v1beta1_runservice.yaml#L240

Reply:
@diviner524 Updated the function. I assume this addresses all scenarios.